src/main/java/com/google/sps/servlets/ProfileBuilderServlet.java

```diff
@@ -0,0 +1,60 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.PersistentUserRepository;
+import com.google.sps.data.User;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for creating new profiles. */
+@WebServlet("/profile")
+public class ProfileBuilderServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    PersistentUserRepository emptyRepo = new PersistentUserRepository();
```

Comment:
consider moving the PersistentUserRepository to class variable rather than creating a new object for each new POST request.

Reply:
+1
Also, emptyRepo as a name isn't necessarily accurate, no? Maybe just 'repository'.

Reply:
Would you like the PersistentUserRepository class variable to be just a variable, or also have it create an object when its initialized as a class variable. 

Ex. 
`
  private static PersistentUserRepository repository = 
    new PersistentUserRepository();`

Reply:
Let's avoid creating the object multiple times by initializing the PersistentUserRepository just once. Your example should work.

Reply:
Ok. Implemented simpleton pattern for this

Reply:
Done